awst-austria/qa4sm-reader
# Required update to handle new QA4SM netCDF structure according to Jir…

…a-744 (#77)

* intial

* started adapting

* adapted existing code to properly handle temporal sub-windows, including renaming the output files and figure titles accordingly

* before testing

* fig titles and filenames now always contain the default or temporal sub-window case's name. also, maps are only created fo r the default case

* intermediate saving

* added all code required to generate comparison boxplots

* fully integrated compariosn plots and all plots now have the logo instead of the watermark

* all statistics .csv files are now stroed in a dedicated zip archive

* intermediate commit

* Fix bugs with metadata plots

* automatic extraction of temporal sub-window names from provided netcdf file and enhanced type hints

* testing of tcol metrics plotting

* removed debugging statements, implemented bug fix for metadata plot generationwit tcol metrics

* comparison boxplots of tcol metrics are not created

* implemented saving of created figures in one go for multiple filetypes

* code for netcdf file transcription to new qa4sm nc structure

* removed accidentally staged restructured files

* implemented means for transcription for all test files

* added automatic netCDF file transcription prior to tests and corrected for temp. sub-win. dimension as additional item in `QA4SMImage.varnames` in `test_image.py::test_load_vars`

* pt2: added automatic netCDF file transcription prior to tests and corrected for temp. sub-win. dimension as additional item in `QA4SMImage.varnames` in `test_image.py::test_load_vars`

* netcdf_transcription.py and intra_annual_temp_windows.py now are part of the `qa4sm_reader` package, isntead of `qa4sm`. Further, the globals file was adapted accordingly and now contains all vars required. `qa4sm` imports these specific vars then from the `qa4sm_reader.globals` file. Lastly, `qa4sm_reader.img.QA4SMImg` was adapted to check if a dimension corresponding to temporal sub-windows exists in the specified netCDF file and if not automatically transcribes the loaded dataset. Tests were adapted accordingly.

* code cleaning.

* code cleaning

* introduced function `qa4sm_reader.utils.filter_out_self_combination_tcmetric_vars()` to exclude TC metric vars of same reference and metric dataset from being plotted

* added pytesmo dependency

* added MANIFEST.in to correctly install the static dir requried for inclusion of logo as watermark

* deleted superflous files

* adapted packaging of reader to incorporat the static dir containing the logo

* tests for intra_annual_temp_windows

* added tests to test_utils and made sure they run first

* started test_netcdf_transcription.py

* fixed test_logo_exists

* fixing test_keep_pytesmo_ncfile

* tests for netcdf_transcription

* finished test development for intra annual metrics and adapation of existing tests

* Update environment.yml (#81)

* Update environment.yml

checking if defining pyscaffold in the env helps fixing the problem with failing tests

* Update CI and dependency list

* Update env

* Update CI

* Update env

* Update CI

---------



* added test to check intra-annual metric related plots and proper file transcription

* fixed tests/test_netcdf_transcription.py::test_correct_file_transcription

* fixed tests/test_netcdf_transcription.py::test_correct_file_transcription

* fixed windows test failing

* fixed mamba workflow issue

* bug fix for  obtaining amount of chars from Path object

---------
